pkg/build/controller/build/build_controller.go

```diff
@@ -2023,39 +2040,42 @@ func (bc *BuildController) getAdditionalTrustedCAData(config *configv1.Image) (m
 	return additionalCA.Data, nil
 }
 
-func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image) (string, error) {
-	registriesConfig := config.Spec.RegistrySources
-	if len(registriesConfig.InsecureRegistries) == 0 && len(registriesConfig.BlockedRegistries) == 0 {
+func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image, policies []*operatorv1alpha1.ImageContentSourcePolicy) (string, error) {
+
+	blockedRegs := []string{}
+	insecureRegs := []string{}
+	if config != nil {
+		insecureRegs = config.Spec.RegistrySources.InsecureRegistries
+		blockedRegs = config.Spec.RegistrySources.BlockedRegistries
+	}
+	if len(insecureRegs) == 0 && len(blockedRegs) == 0 && len(policies) == 0 {
 		klog.V(4).Info("using default registry settings for builds")
 		return "", nil
 	}
-	configObj := tomlConfig{
-		Registries: registries{
-			// docker.io must be the only entry in the registry search list
-			// See https://github.com/openshift/builder/pull/40
-			Search: registryList{
-				Registries: []string{"docker.io"},
-			},
-			Insecure: registryList{
-				Registries: registriesConfig.InsecureRegistries,
-			},
-			Block: registryList{
-				Registries: registriesConfig.BlockedRegistries,
-			},
-		},
-	}
 
-	configTOML, err := toml.Marshal(configObj)
+	configObj := sysregistriesv2.V2RegistriesConf{}
+	// docker.io must be the only entry in the registry search list
+	// See https://github.com/openshift/builder/pull/40
+	configObj.UnqualifiedSearchRegistries = []string{"docker.io"}
+	err := mco.EditRegistriesConfig(&configObj, insecureRegs, blockedRegs, policies)
 	if err != nil {
+		klog.V(4).Infof("MCO library had problem building registries config: %s", err.Error())
```

Comment:
Use a higher log level for an error - `V(0)`?

Reply:
sure switched to v0